engine/schema/src/main/java/com/cloud/user/dao/AccountDaoImpl.java

```diff
@@ -157,9 +157,16 @@ public AccountDaoImpl() {
 
     @Override
     public List<AccountVO> findAccountsLike(String accountName) {
+        return findAccountsLike(accountName, null);
+    }
+
+    @Override
+    public List<AccountVO> findAccountsLike(String accountName, Filter filter) {
         SearchCriteria<AccountVO> sc = createSearchCriteria();
-        sc.addAnd("accountName", SearchCriteria.Op.LIKE, "%" + accountName + "%");
-        return listBy(sc);
+        if (accountName != null) {
```

Comment:
Why not use `Optional` instead of null?

Reply:
Doesn't it inherently do the same thing ?

Reply:
@davidjumani Optional [1] allows avoiding NPEs while gives a way to check if say a String is present, null etc. Your check is okay, but maybe add a `Strings.isNullOrEmpty` instead to avoid search criteria clause for empty accountName.
[1] https://docs.oracle.com/javase/8/docs/api/java/util/Optional.html

Reply:
> Doesn't it inherently do the same thing ?

It inherently does the same thing but using optionals is preferred over checking explicitly for `null`